action.py: No such file or directory

**Description**

I keep getting an error using the most barebones configuration, I will skip to the part when the runner finishes installing all dependencies and tries to run the sigstore action.

```
Run D:\a\_actions\sigstore\gh-action-sigstore-python\v1.2.3/action.py "./build/test.dist/test.exe"
D:\a\_temp\4cbeed94-1ea6-46c1-a605-345eda4ef060.sh: line 1: D:a_actionssigstoregh-action-sigstore-pythonv1.2.3/action.py: No such file or directory
Error: Process completed with exit code 127.
```
Note: Isn't this `D:a_actionssigstoregh-action-sigstore-pythonv1.2.3/action.py` path kind of broken? I feel like it's missing a backslash or two. Maybe this action is not suitable for a windows runner?

I tested on this workflow.

```yaml
name: Python application

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

permissions:
  contents: read
  id-token: write

jobs:

  build:
    # Windows is currently the only platform Nuitka action supports apparently?
    runs-on: windows-latest

    steps:

      # Check-out repository
      - uses: actions/checkout@v3

      # Setup Python
      - uses: actions/setup-python@v4
        with:
          python-version: '3.10' 
          architecture: 'x64' 

      - name: Compile to executable
        uses: Nuitka/Nuitka-Action@main
        with:
          nuitka-version: main
          script-name: test.py
          standalone: true
          onefile: false


      - name: Sign
        uses: sigstore/gh-action-sigstore-python@v1.2.3
        with:
          inputs: build/test.dist/test.exe 

      # Uploads artifact
      - name: Upload Artifact
        uses: actions/upload-artifact@v3
        with:
          name: Windows Build
          path: build/test.dist
```
Note: The input path seems to be correct, as removing the whole signing step uploads the artifact correctly.

**Version**

gh-action-sigstore-python@v1.2.3
windows-latest

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -13,6 +13,6 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: "3.7"
+          python-version: "3.x"
       - name: lint
         run: make lint

--- a/.github/workflows/selftest.yml
+++ b/.github/workflows/selftest.yml
@@ -33,6 +33,7 @@ jobs:
         with:
           inputs: ./test/artifact.txt
       - name: Check outputs
+        shell: bash
         run: |
           [[ -f ./test/artifact.txt.sigstore ]] || exit 1
 

--- a/.github/workflows/selftest.yml
+++ b/.github/workflows/selftest.yml
@@ -23,6 +23,10 @@ jobs:
     if: (github.event_name != 'pull_request') || !github.event.pull_request.head.repo.fork
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        if: ${{ matrix.os != 'ubuntu-latest' }}
+        with:
+          python-version: "3.x"
       - name: Sign artifact and publish signature
         uses: ./
         id: sigstore-python

--- a/.github/workflows/selftest.yml
+++ b/.github/workflows/selftest.yml
@@ -13,7 +13,13 @@ permissions:
 
 jobs:
   selftest:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - macos-latest
+          - windows-latest
+    runs-on: ${{ matrix.os }}
     if: (github.event_name != 'pull_request') || !github.event.pull_request.head.repo.fork
     steps:
       - uses: actions/checkout@v3

--- a/action.py
+++ b/action.py
@@ -278,12 +278,15 @@ if sign_status.returncode != 0:
 _github_env = os.getenv("GITHUB_ENV")
 assert _github_env is not None
 with Path(_github_env).open("a") as gh_env:
-    delim = os.urandom(16).hex()
     # Multiline values must match the following syntax:
     #
-    # {name}<<{delimiter}
-    # {value}
-    # {delimiter}
+    #  {name}<<{delimiter}
+    #  {value}
+    #  {delimiter}
+    #
+    # We use a random delimiter to avoid potential conflicts with our input;
+    # see: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions
+    delim = os.urandom(16).hex()
     print(f"GHA_SIGSTORE_PYTHON_INTERNAL_SIGNING_ARTIFACTS<<{delim}", file=gh_env)
     print("\n".join(signing_artifact_paths), file=gh_env)
     print(delim, file=gh_env)

--- a/action.py
+++ b/action.py
@@ -20,6 +20,7 @@
 # is a whitespace-separated list of inputs
 
 import os
+import random
 import string
 import subprocess
 import sys
@@ -278,19 +279,15 @@ if sign_status.returncode != 0:
 _github_env = os.getenv("GITHUB_ENV")
 assert _github_env is not None
 with Path(_github_env).open("a") as gh_env:
+    delim = random.randbytes(16).hex()
     # Multiline values must match the following syntax:
     #
     # {name}<<{delimiter}
     # {value}
     # {delimiter}
-    print(
-        "GHA_SIGSTORE_PYTHON_INTERNAL_SIGNING_ARTIFACTS<<EOF"
-        + os.linesep
-        + os.linesep.join(signing_artifact_paths)
-        + os.linesep
-        + "EOF",
-        file=gh_env,
-    )
+    print(f"GHA_SIGSTORE_PYTHON_INTERNAL_SIGNING_ARTIFACTS<<{delim}", file=gh_env)
+    print("\n".join(signing_artifact_paths))
+    print(delim, file=gh_env)
 
 
 # If signing didn't fail, then we check the verification status, if present.

--- a/action.py
+++ b/action.py
@@ -20,7 +20,6 @@
 # is a whitespace-separated list of inputs
 
 import os
-import random
 import string
 import subprocess
 import sys
@@ -279,7 +278,7 @@ if sign_status.returncode != 0:
 _github_env = os.getenv("GITHUB_ENV")
 assert _github_env is not None
 with Path(_github_env).open("a") as gh_env:
-    delim = random.randbytes(16).hex()
+    delim = os.urandom(16).hex()
     # Multiline values must match the following syntax:
     #
     # {name}<<{delimiter}

--- a/action.py
+++ b/action.py
@@ -286,7 +286,7 @@ with Path(_github_env).open("a") as gh_env:
     # {value}
     # {delimiter}
     print(f"GHA_SIGSTORE_PYTHON_INTERNAL_SIGNING_ARTIFACTS<<{delim}", file=gh_env)
-    print("\n".join(signing_artifact_paths))
+    print("\n".join(signing_artifact_paths), file=gh_env)
     print(delim, file=gh_env)
 
 

--- a/action.py
+++ b/action.py
@@ -283,12 +283,13 @@ with Path(_github_env).open("a") as gh_env:
     # {name}<<{delimiter}
     # {value}
     # {delimiter}
-    gh_env.write(
+    print(
         "GHA_SIGSTORE_PYTHON_INTERNAL_SIGNING_ARTIFACTS<<EOF"
         + os.linesep
         + os.linesep.join(signing_artifact_paths)
         + os.linesep
-        + "EOF"
+        + "EOF",
+        file=gh_env,
     )
 
 

--- a/action.yml
+++ b/action.yml
@@ -109,6 +109,8 @@ runs:
       run: |
         ${GITHUB_ACTION_PATH}/action.py "${GHA_SIGSTORE_PYTHON_INPUTS}"
       env:
+        # The year is 2023, and nonsense like this is still necessary on Windows.
+        PYTHONUTF8: "1"
         GHA_SIGSTORE_PYTHON_IDENTITY_TOKEN: "${{ inputs.identity-token }}"
         GHA_SIGSTORE_PYTHON_SIGNATURE: "${{ inputs.signature }}"
         GHA_SIGSTORE_PYTHON_CERTIFICATE: "${{ inputs.certificate }}"

--- a/action.yml
+++ b/action.yml
@@ -108,6 +108,7 @@ runs:
       id: sigstore-python
       run: |
         ${GITHUB_ACTION_PATH}/action.py "${GHA_SIGSTORE_PYTHON_INPUTS}"
+      shell: bash
       env:
         GHA_SIGSTORE_PYTHON_IDENTITY_TOKEN: "${{ inputs.identity-token }}"
         GHA_SIGSTORE_PYTHON_SIGNATURE: "${{ inputs.signature }}"

--- a/action.yml
+++ b/action.yml
@@ -101,13 +101,13 @@ runs:
     - name: Set up sigstore-python
       run: |
         # NOTE: Sourced, not executed as a script.
-        source "${{ github.action_path }}/setup/setup.bash"
+        source "${GITHUB_ACTION_PATH}/setup/setup.bash"
       shell: bash
 
     - name: Run sigstore-python
       id: sigstore-python
       run: |
-        ${{ github.action_path }}/action.py "${{ inputs.inputs }}"
+        ${GITHUB_ACTION_PATH}/action.py "${GHA_SIGSTORE_PYTHON_INPUTS}"
       env:
         GHA_SIGSTORE_PYTHON_IDENTITY_TOKEN: "${{ inputs.identity-token }}"
         GHA_SIGSTORE_PYTHON_SIGNATURE: "${{ inputs.signature }}"
@@ -125,6 +125,7 @@ runs:
         GHA_SIGSTORE_PYTHON_VERIFY_OIDC_ISSUER: "${{ inputs.verify-oidc-issuer }}"
         GHA_SIGSTORE_PYTHON_RELEASE_SIGNING_ARTIFACTS: "${{ inputs.release-signing-artifacts }}"
         GHA_SIGSTORE_PYTHON_INTERNAL_BE_CAREFUL_DEBUG: "${{ inputs.internal-be-careful-debug }}"
+        GHA_SIGSTORE_PYTHON_INPUTS: "${{ inputs.inputs }}"
       shell: bash
 
     - uses: actions/upload-artifact@v3

--- a/action.yml
+++ b/action.yml
@@ -108,7 +108,6 @@ runs:
       id: sigstore-python
       run: |
         ${GITHUB_ACTION_PATH}/action.py "${GHA_SIGSTORE_PYTHON_INPUTS}"
-      shell: bash
       env:
         GHA_SIGSTORE_PYTHON_IDENTITY_TOKEN: "${{ inputs.identity-token }}"
         GHA_SIGSTORE_PYTHON_SIGNATURE: "${{ inputs.signature }}"

